geo/src/algorithm/scale.rs

```diff
@@ -1,49 +1,150 @@
-use crate::{AffineTransform, CoordNum, MapCoords, MapCoordsInPlace, Point};
+use crate::{AffineOps, AffineTransform, BoundingRect, CoordFloat, CoordNum, Coordinate, Rect};
 
 /// An affine transformation which scales a geometry up or down by a factor.
 ///
-/// The point of origin is *usually* given as the 2D bounding box centre of the geometry, but
-/// any coordinate may be specified.
-///
 /// ## Performance
 ///
 /// If you will be performing multiple transformations, like [`Scale`](crate::Scale),
 /// [`Skew`](crate::Skew), [`Translate`](crate::Translate), or [`Rotate`](crate::Rotate), it is more
 /// efficient to compose the transformations and apply them as a single operation using the
 /// [`AffineOps`](crate::AffineOps) trait.
-///
-/// # Examples
-///
-/// ```
-/// use geo::Scale;
-/// use geo::{line_string, BoundingRect, LineString};
-///
-/// let ls: LineString<f64> = line_string![
-///     (x: 0.0f64, y: 0.0f64),
-///     (x: 10.0f64, y: 10.0f64),
-/// ];
-/// let origin = ls.bounding_rect().unwrap().center();
-/// let scaled = ls.scale(10.0, 10.0, origin.into());
-///
-/// assert_eq!(scaled, line_string![
-///     (x: -45.0f64, y: -45.0f64),
-///     (x: 55.0f64, y: 55.0f64)
-/// ]);
-/// ```
-///
-pub trait Scale<T> {
-    fn scale(&self, x: T, y: T, origin: Point<T>) -> Self
-    where
-        T: CoordNum;
+pub trait Scale<T: CoordNum> {
+    /// Scale a geometry from it's bounding box center.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::Scale;
+    /// use geo::{LineString, line_string};
+    ///
+    /// let ls: LineString = line_string![(x: 0., y: 0.), (x: 10., y: 10.)];
+    ///
+    /// let scaled = ls.scale(2.);
+    ///
+    /// assert_eq!(scaled, line_string![
+    ///     (x: -5., y: -5.),
+    ///     (x: 15., y: 15.)
+    /// ]);
+    /// ```
+    #[must_use]
+    fn scale(&self, scale_factor: T) -> Self;
```

Comment:
This might be controversial, so I wanted to call it out:

I've re-purposed the nice `scale` name for the method I think people will use the most. 

The general purpose method with all the flexible parameters (which I believe will be less commonly used) is now called `scale_around_point`.

WDYT?

Reply:
Yep I agree this is an ergonomic improvement!